core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -93,18 +93,107 @@ func generateFungibleAssetLockContractId(ctx contractapi.TransactionContextInter
 	return contractId
 }
 
+/*
+ * Function to validate the locker in asset agreement.
+ * If locker is not set, it will be set to the caller.
+ * If the locker is set already, it ensures that the locker is same as the creator of the transaction.
+ */
+func validateAndSetLockerOfAssetAgreement(ctx contractapi.TransactionContextInterface, assetAgreement *common.AssetExchangeAgreement) error {
```

Comment:
You can probably just pass `Locker` instead of the whole agreement here, so this function can be called by the locking functions that rely on contract IDs.

Reply:
This is not possible since we are also setting `Locker` in case it was not set earlier.